packages/host/app/components/ai-assistant/panel.gts

```diff
@@ -53,6 +53,14 @@ interface Signature {
   };
 }
 
+export interface SessionRoomData {
```

Comment:
I'm wondering why this another layer of data (`SessionRoomData`) is needed. I think the point of having a model is to hold all the room related data so you wouldn't need additional structures

Reply:
This is the "Model" for session-related stuff. Perhaps, this will be a little clearer if I share possible future plans. So, one thing I considered is that `roomResource` should be a resource of the room and not be concerned with session data. We currently, bottle everything up into one by using a map of roomResources. I don't actually believe in this and the patterns seem a little awkard. We should have ANOTHER resource which stores internally a map of roomId to session data that can be reindexed -- but not whole messages just recent ones and possibly have other things like membership too

Reply:
> How come `roomId` and `name` are in this structure separately? Isn't the purpose of `RoomModel` to have this data? Also `created`

I made a suggestion below to essentially convert RoomModel to RoomEvents

Reply:
Btw appreciate this feedback. I do agree when we had a single room field it was somewhat clearer bcos everything was in one place. But, I think based upon where we are going we should eliminate RoomModel all together.

Reply:
@jurgenwerk does this work https://github.com/cardstack/boxel/pull/1424. Is it clearer?